week2-section2/exercise2/Spotify.cpp

```diff
@@ -0,0 +1,181 @@
+		
+		//adding the song in playlist
+		Playlist.AddSong(Song);
+
+		//menu
+		std::cout << "1 - Enter another song\n2 - Go back to main menu" << std::endl;
+		int MenuSelection;
+		std::cin >> MenuSelection;
+		while (std::cin.fail() || (MenuSelection < 1) || (MenuSelection > 2))
+		{
+			std::system("cls");
+			std::cin.clear();
+			std::cin.ignore(100, '\n');
+			std::cout << "Invalid Option, Insert a valid option" << std::endl;
+			std::cout << "1 - Enter another song\n2 - Go back to main menu" << std::endl;
+			std::cin >> MenuSelection;
+		}
+		if (MenuSelection == 2)
+		{
+			//adding the playlist in container
+			PlaylistContainer.AddPlaylist(Playlist);
+			bWantToAddASong = false;
+			FSpotify::OpenMainMenu();
+		}
+		else if (MenuSelection == 1) {
+			std::system("cls");
+			bWantToAddASong = true;
+		}
+	}
+};
+
+//----------------------------------------------------------
+void FSpotify::OpenBrowsePlaylistsMenu() 
+{
+	std::system("cls");
+	std::cout << "Here are your playlist(s):" << std::endl;
+	
+	//Looking the playlists saved
+	PlaylistContainer.GetCurrentPlaylistCount();
+	std::cout << "\nWhich playlist do you want to see: " << std::endl;
+
+	//choosing the playlist
+	int InputConverter;
+	std::cin >> InputConverter;
+	while (std::cin.fail())
+	{
+		std::cin.clear();
+		std::cin.ignore(100, '\n');
+		std::cout << "Invalid Option " << std::endl;
+
+		std::system("cls");
+		std::cout << "Here are your playlist(s):" << std::endl;
+
+		//Looking the playlists saved
+		PlaylistContainer.GetCurrentPlaylistCount();
+		std::cout << "\nWhich playlist do you want to see: " << std::endl;
+
+		std::cin >> InputConverter;
+	}
+	std::system("cls");
+	PlaylistContainer.GetPlaylist(InputConverter - 1);
+
+	//Menu
+	std::cout << "1 - Pick another playlist\n2 - Go back to main menu" << std::endl;
+	int MenuSelection;
+	std::cin >> MenuSelection;
+	while (std::cin.fail() || (MenuSelection < 1) || (MenuSelection > 2))
+	{
+		std::system("cls");
+		std::cin.clear();
+		std::cin.ignore(100, '\n');
+		std::cout << "Invalid Option, Insert a valid option" << std::endl;
+		std::cout << "1 - Pick another playlist\n2 - Go back to main menu" << std::endl;
+		std::cin >> MenuSelection;
+	}
+	if (MenuSelection == 1)
+	{
+		std::system("cls");
+		FSpotify::OpenBrowsePlaylistsMenu();
+	}
+	else
+	{
+		std::system("cls");
+		FSpotify::OpenMainMenu();
+	}
+};
+
+//-----------------------------------------------------------------------
+int FSpotify::RunApp() 
+{
+	while (ActiveMenuCommand != FSpotify::EMenuCommand::Exit)
+	{
+		if (ActiveMenuCommand == FSpotify::EMenuCommand::MainMenu)
+		{
+			FSpotify::OpenMainMenu();
+		}
+		else if (ActiveMenuCommand == FSpotify::EMenuCommand::CreatePlaylist)
+		{
+			FSpotify::OpenCreatePlaylistMenu();
+		}
+		else if (ActiveMenuCommand == FSpotify::EMenuCommand::BrowsePlaylists)
+		{
+			FSpotify::OpenBrowsePlaylistsMenu();
+		}
+	}
+	if (ActiveMenuCommand == FSpotify::EMenuCommand::Exit)
```

Comment:
Also, you wouldn't need an if for this. Remember that this part of the code (after the while loop) is only executed if the condition in the while loop fails, so this wouldn't be executed if ActiveMenuCommand wasn't Exit.

Reply:
Also, please remove the unnecessary `FSpotify::` uses in your code.